spring-cloud-kubernetes-fabric8-discovery/src/main/java/org/springframework/cloud/kubernetes/fabric8/discovery/Fabric8DiscoveryServicesAdapter.java

```diff
@@ -0,0 +1,87 @@
+/*
+ * Copyright 2012-2022 the original author or authors.
+ *
+ * Licensed under the Apache License, Version 2.0 (the "License");
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ *      https://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+package org.springframework.cloud.kubernetes.fabric8.discovery;
+
+import java.util.ArrayList;
+import java.util.List;
+import java.util.Optional;
+import java.util.function.Function;
+import java.util.function.Predicate;
+
+import io.fabric8.kubernetes.api.model.Service;
+import io.fabric8.kubernetes.client.KubernetesClient;
+
+import org.springframework.cloud.kubernetes.commons.discovery.KubernetesDiscoveryProperties;
+import org.springframework.expression.Expression;
+import org.springframework.expression.spel.standard.SpelExpressionParser;
+import org.springframework.expression.spel.support.SimpleEvaluationContext;
+
+/**
+ * Adapts a {@link KubernetesClientServicesFunction} to a Function that takes a
+ * KubernetesClient as input and returns a List of Services(s), plus adds functionality
+ * not supported by it.
+ *
+ * @author wind57
+ */
+final class Fabric8DiscoveryServicesAdapter implements Function<KubernetesClient, List<Service>> {
+
+	private static final SpelExpressionParser PARSER = new SpelExpressionParser();
+
+	private static final SimpleEvaluationContext EVALUATION_CONTEXT = SimpleEvaluationContext.forReadOnlyDataBinding()
+			.withInstanceMethods().build();
+
+	private final KubernetesClientServicesFunction function;
```

Comment:
the idea of having the `KubernetesClientServicesFunction` was probably good in the beginning when the code was only taking shape, now though, with the introduction of only some namespaces, is not the case anymore. 

Instead of removing it, I though of creating this "adapter" that allows to still use it, but at the same time allow the computation of services based on multiple namespaces.